sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -45,25 +44,36 @@ public static AmqpMessage BatchSBMessagesAsAmqpMessage(IEnumerable<SBMessage> so
         ///
         /// <returns>The batch <see cref="AmqpMessage" /> containing the source messages.</returns>
         ///
-        private static AmqpMessage BuildAmqpBatchFromMessage(IEnumerable<SBMessage> source, bool forceBatch)
+        private static AmqpMessage BuildAmqpBatchFromMessage(IEnumerable<ServiceBusMessage> source, bool forceBatch)
         {
             AmqpMessage firstAmqpMessage = null;
-            SBMessage firstMessage = null;
-
-            return BuildAmqpBatchFromMessages(
-                source.Select(sbMessage =>
+            ServiceBusMessage firstMessage = null;
+            List<AmqpMessage> amqpMessagesForBatching = new List<AmqpMessage>();
+            try
+            {
+                foreach (var sbMessage in source)
                 {
                     if (firstAmqpMessage == null)
                     {
                         firstAmqpMessage = SBMessageToAmqpMessage(sbMessage);
                         firstMessage = sbMessage;
-                        return firstAmqpMessage;
+                        amqpMessagesForBatching.Add(firstAmqpMessage);
                     }
                     else
                     {
-                        return SBMessageToAmqpMessage(sbMessage);
+                        amqpMessagesForBatching.Add(SBMessageToAmqpMessage(sbMessage));
                     }
-                }).ToList(), firstMessage, forceBatch);
+                }
+
+                return BuildAmqpBatchFromMessages(amqpMessagesForBatching, firstMessage, forceBatch);
+            }
+            finally
+            {
+                foreach (var amqpMessage in amqpMessagesForBatching)
```

Comment:
I don't think we can dispose these since the underlying buffer gets used in the AmqpMessage that is sent.

Reply:
When adding messages to a `ServiceBusMessageBatch`, we create an envelope message and dispose that - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageBatch.cs#L133

I _think_ that this would handle properly disposing the buffer used by the actual message that is added to the envelope.

Reply:
Yeah I missed that.